index.js

```diff
@@ -156,6 +156,7 @@ var extensions = {
         });
       },
     },
+    '@swc/register',
```

Comment:
@HRKings I believe we need to implement this the same way that babel is registered. Otherwise swc hooks **all** of their DEFAULT_EXTENSIONS. We only want to hook `.ts` which can be passed as an option. https://github.com/swc-project/register/blob/master/src/node.ts#L106

Reply:
It looks like they support the `ignore` option, too. So it'll be almost the same, without `rootMode`

Reply:
swc might only support regexp strings in the ignore, I can't tell and couldn't find it in their docs. Their code is also very hard to traverse so idk.

Reply:
Sorry for the delay, I added the hook for swc

Reply:
@HRKings I'm just waiting on a new release from the swc/register team to get support for this: https://github.com/swc-project/register/commit/d9a617ad646976dcf9b5735c2ef824f2bdb8bb28

Fwiw, I think this will allow us to implement `.swc.js` like we do `.babel.js` for babel.

Reply:
It seems that they released it; however, the author doesn't understand why programmatic usage would be useful so we have to import it as `@swc/register/lib/node`